.devcontainer/scripts/post-creation.sh

```diff
@@ -0,0 +1,4 @@
+# Install SDK and tool dependencies before container starts
+# Also run the full restore on the repo so that go-to definition
+# and other language features will be available in C# files
+./restore.sh
```

Comment:
For this command, I think you may also need to run eng/sdk-build-env.sh as well to ensure everything is set up correctly for VSCode.

Reply:
Thanks, will have a check and test

Reply:
@baronfel , I did not find the `eng/sdk-build-env.sh` file, is there a typo?

![image](https://github.com/dotnet/sdk/assets/7604648/7833d51f-196b-477c-b6e7-45d0c773b6b1)

Reply:
It's created during the build.sh/restore.sh calls:

https://github.com/dotnet/sdk/blob/main/eng%2Frestore-toolset.sh

It sets up the environment variables for the .NET SDK to force it to use the just-downloaded SDK and Runtime(s).

Reply:
Seemed file not found from my test
![image](https://github.com/dotnet/sdk/assets/7604648/f6a55791-9b98-4a53-94a2-648eac4950b3)

Reply:
ah, that's because only the `build` command in the build.sh script calls the generation. I'd suggest that instead of calling `restore.sh` in your post-creation.sh script, you should call `./build.sh`. This will be the most likely way to get a usable installation, in my opinion.

Reply:
Ah, I found it in `artifacts/sdk-build-env.sh`, the `restore.sh`/`build.sh` would produce the script both.

From the script content, seemed there's duplicate `/` in the dotnet path, does this matter?
![image](https://github.com/dotnet/sdk/assets/7604648/70d61ab7-70fd-4b93-85be-d8247d682ecc)

It may come from 
https://github.com/dotnet/sdk/blob/a5caad8ad13f3564574f6ac459f3ed43320cb97f/eng/common/tools.sh#L144-L146

Reply:
it _shouldn't_ matter, but it's worth fixing. good spot!

Reply:
from my test, the `sdk-build-env.sh` would not keep effective when execting, any special flag when executing?

![image](https://github.com/dotnet/sdk/assets/7604648/51839aeb-fcbc-4f0d-9048-2d63f43e1e77)

think maybe it's not a need to execute, or should I configure the environment variables in the devcontainer config like `PATH` do https://github.com/dotnet/aspnetcore/blob/f26b79eb34d13648bedc5ec98e2151f8c55db978/.devcontainer/devcontainer.json#L33

Reply:
That's bizarre. If the script isn't working, I think setting the environment variables explicitly like aspnetcore is a fine way to go. Thanks for taking the time to dig into this.